src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -159,6 +159,13 @@ internal ODataQueryContext QueryContext
         /// </summary>
         public ODataPath Path { get; set; }
 
+        internal Type Type { get; set; }
+
+        internal bool IsUntyped { get { return typeof(IEdmObject).IsAssignableFrom(Type); } }
+
+        internal bool IsDeltaOfT { get { return Type != null && TypeHelper.IsGenericType(Type) && (Type.GetGenericTypeDefinition() == typeof(DeltaSet<>) || 
+                    Type.GetGenericTypeDefinition() == typeof(Delta<>) || Type.GetGenericTypeDefinition() == typeof(DeltaDeletedEntityObject<>)); } }
```

Comment:
this is really expensive. calculate this the first time it's requested and then cache until Type is reset.